app/src/main/java/com/teamagam/gimelgimel/app/view/viewer/cesium/CesiumMapView.java

```diff
@@ -264,6 +273,42 @@ public boolean isReady() {
         return mIsGGMapReadySynchronized.getData();
     }
 
+    @Override
+    public void saveViewState(final Bundle outState) {
+        if(outState != null) {
+            outState.putParcelable(CURRENT_CAMERA_POSITION_KEY, getLastViewedLocation());
+        }
+    }
+
+    @Override
+    public void restoreViewState(Bundle inState) {
+        final PointGeometry savedLocation = inState.getParcelable(CURRENT_CAMERA_POSITION_KEY);
+
+        // Make sure the Bundle isn't null.
+        // Null is a valid value, because the Bundle can be null in some situations,
+        // and throwing an exception here will crash the app,
+        // instead of initialize with default values.
+        if(inState != null) {
+            // Also, check the savedLocation object, the bundle may return null or default,
+            // if the save hasn't occurred yet.
+            if(savedLocation != null && savedLocation != PointGeometry.DEFAULT_POINT) {
+                mLocationUpdater.setLastViewedLocation(savedLocation);
+
+                // Wait for the map to be ready before zooming into the last view.
```

Comment:
Extract all the below code-block into a method?
Perhaps, restoreMapExtent(PointGeometry location)